Update contribution guidelines (#2127)
Human-Connection/Human-Connection: CONTRIBUTING.md

Every pull request needs to:
* fix an issue (if there is something you want to work on but there is no issue for it, create one first and discuss it with the team)
* include tests for the code that is added or changed
* pass all tests (linter, backend, frontend, end-to-end)
* be approved by at least 1 developer who is not the owner of the PR (when more than 10 files were changed it needs 2 approvals)

## The Team

There are many volunteers all around the world helping us build this network and without their contributions we wouldn't be where we are today. Big thank you to all of you!

You can see the core team behind Human Connection [on our website](https://human-connection.org/en/the-team/). On Github you will mostly run into our developers:
* Robert (@roschaefer)
* Matt (@mattwr18)
* Wolle (@Tirokk)
* Alex (@ogerly)
* Alina (@alina-beck)
* Martin (@datenbrei), our head of IT
* and sometimes Dennis (@DennisHack), the founder of Human Connection

## Meetings and Pair Programming Sessions

Times below refer to **German Time** – that's CET (GMT+1) in winter and CEST (GMT+2) in summer – because most Human Connection core team members are living in Germany.

Daily standup

> * every Monday–Friday 11:45

Reviewer: ```suggestion
* every Monday–Friday 11:30
```

Author: nooo, daily is at 11:45 – or am I just always 15 minutes late? 😜

Reviewer 2: we agreed to change it since we are always rushing to get to the 12pm meeting @alina-beck

Author: oh, I had no idea! I'll merge this now, thanks for updating :)

# CONTRIBUTING

Thank you so much for thinking of contributing to the Human Connection project! It's awesome you're here, we really appreciate it. :-\)

## Getting Set Up

Instructions for how to install all the necessary software and some code guidelines can be found in our [documentation](https://docs.human-connection.org/human-connection/).

> To get you started we recommend that you join forces with a regular contributor. Please join [our discord instance](https://discord.gg/6ub73U3) to chat with developers or just get in touch directly on an issue on either [Github](https://github.com/Human-Connection/Human-Connection/issues) or [Zenhub](https://app.zenhub.com/workspaces/human-connection-nitro-5c0154ecc699f60fc92cf11f/boards?repos=152252353):

Reviewer: ```suggestion
To get you started we recommend that you join forces with a regular contributor. Please join [our discord instance](https://human-connection.org/discord) to chat with developers or just get in touch directly on an issue on either [Github](https://github.com/Human-Connection/Human-Connection/issues) or [Zenhub](https://app.zenhub.com/workspaces/human-connection-nitro-5c0154ecc699f60fc92cf11f/boards?repos=152252353):
```

Reviewer: For reasons, the discord invite link got lost sometimes in the past. This redirect should always lead to a valid invite link.

2. We communicate with the team to see if the issue is still available. (When you comment on an issue but don't get an answer there within 1-2 days try to mention @Human-Connection/hc-dev-team to make sure we check in.)
3. We make sure we understand the issue in detail – what problem is it solving and how should it be implemented?
4. We assign ourselves to the issue and move it to `In Progress` on [Zenhub](https://app.zenhub.com/workspaces/human-connection-nitro-5c0154ecc699f60fc92cf11f).
5. We start working on it in a `new branch` and open a `pull request` prefixed with `[WIP]` (work in progress) to which we regularly push our changes.
6. When questions come up we clarify them with the team (directly in the issue on Github).
7. When we are happy with our work and our PR is passing all tests we remove the `[WIP]` from the PR description and ask for reviews (if you're not sure who to ask there is @Human-Connection/hc-dev-team which pings all core developers).
8. We then incorporate the suggestions from the reviews into our work and once it has been approved it can be merged into master!

Every pull request needs to:

> * fix an issue (if there is something you want to work on but there is no issue for it, create on first and discuss it with the team)

Reviewer: ```suggestion
* fix an issue (if there is something you want to work on but there is no issue for it, create one first and discuss it with the team)
```

* include tests for the code that is added or changed
* pass all tests (linter, backend, frontend, end-to-end)
* be approved by at least 1 developer who is not the owner of the PR (when more than 10 files were changed it needs 2 approvals)

## The Team

There are many volunteers all around the world helping us build this network and without their contributions we wouldn't be where we are today. Big thank you to all of you!

You can see the core team behind Human Connection [on our website](https://human-connection.org/en/the-team/). On Github you will mostly run into our developers:
* Robert (@roschaefer)
* Matt (@mattwr18)
* Wolle (@Tirokk)
* Alex (@ogerly)
* Alina (@alina-beck)
* Martin (@datenbrei), our head of IT
* and sometimes Dennis (@DennisHack), the founder of Human Connection

## Meetings and Pair Programming Sessions

Daily standup

> * every Monday–Friday 10:45 UTC

Reviewer: Publishing those events on UTC means it's a reference. So, if Germany has its clocks shifted from CET to CEST, our meeting in local time will change from 11:45 CET to 10:45 CEST. Are we OK with that? @mattwr18 @datenbrei @Tirokk @ogerly @alina-beck ?

Reviewer 3: Yes. Also I'd add the place to the standup information (Discord Voicechat -> Coneference Room)

Author: Times are now indicated in "german time" – it's not perfect but clear enough, I hope?

* in the discord `Conference Room`
* all contributors welcome!
* everybody shares what they are working on and asks for help if they are blocked

Regular pair programming sessions
* every Monday, Wednesday and Thursday 14:00 UTC
* the link will be posted in the [discord chat](https://discord.gg/6ub73U3) and on the [Agile Ventures website](https://www.agileventures.org/events?utf8=%E2%9C%93&project_id=220&commit=Filter+by+Project)
* all contributors welcome!
* we team up and work on an issue together (often using Visual Studio live sharing sessions)

> Developer's Community Meeting

Reviewer: ```suggestion
Open-Source Community Meeting
```
It's a little more inclusive. Sometimes people want to join who are no developers.

> * every Thursday 12:00 UTC

Reviewer: When I see English text and a 12:00 meeting I always wonder: 12:00am, 12:00pm? What's actually 12:00pm? Noon, midnight? :thinking: 

From the context I can tell that you are using 24 hours clock here (which I prefer for the reasons above :point_up: ) but it always gives me headache.

Author: It's sooo confusing... luckily we don't have any meetings at 12:00 German time! 😝

Reviewer 2: 12pm is midday... 12am is midnight :stuck_out_tongue:

* the link will be posted in the [discord chat](https://discord.gg/6ub73U3) and on the [Agile Ventures website](https://www.agileventures.org/events?utf8=%E2%9C%93&project_id=220&commit=Filter+by+Project)
* all contributors welcome!

> * ???

Reviewer: ```suggestion
```

Meet the team

> * every Monday 20:00 UTC (at the moment only in German)

Reviewer: ```suggestion
* every Monday 20:00 UTC (at the moment only in German)
* details here https://human-connection.org/veranstaltungen/
```

* via this [zoom link](https://zoom.us/j/936943532)
* all contributors and users of the network welcome!
* users of the network chat with the Human Connection team and discuss current questions and issues

Sprint planning
* bi-weekly on Tuesday 12:00 UTC
* via this [zoom link](https://zoom.us/j/7743582385)
* all contributors welcome (recommended for those who want to work on an issue in this sprint)
* we select and prioritise the issues we will work on in the following two weeks

Sprint retrospective
* bi-weekly on Monday 12:00 UTC
* via this [zoom link](https://zoom.us/j/7743582385)
* all contributors welcome (most interesting for those who participated in the sprint)
* we review the past sprint and talk about what went well and what we could improve

## Philosophy

We practise [collective code ownership](http://www.extremeprogramming.org/rules/collective.html) rather than strong code ownership, which means that:
* developers can make contributions to other people's PRs (after checking in with them)
* we avoid blocking because someone else isn't working, so we sometimes take over PRs from other developers
* everyone should always push their code to branches so others can see it

> 

Reviewer: Stange that these notes found their way into `CONTRIBUTING.md`
